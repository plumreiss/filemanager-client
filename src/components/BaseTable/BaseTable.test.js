import { getByText, render } from "@testing-library/react";
import { BaseTable } from ".";

describe("<BaseTable />", () => {
  it("Should render correctly", () => {
    const lines = [
      {
        text: "uzUVbPfNKGawHWZHeqiO",
        number: 777459,
        hex: "1ec4c6a048194c9b72fad4b0273f2533",
      },
      {
        text: "angel",
        number: 888,
        hex: "4f9e5d1c3a7b2eca8fb3456789012345",
      },
    ];

    const component = render(
      <BaseTable
        files={[
          {
            file: "test2.csv",
            lines,
          },
        ]}
      />
    );

    lines.forEach(({ text, number, hex }) => {
      component.getByText(text);
      component.getByText(number);
      component.getByText(hex);
    });

    expect(component.container).toMatchSnapshot();
  });
});
