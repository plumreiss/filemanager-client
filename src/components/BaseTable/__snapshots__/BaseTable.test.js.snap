// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseTable /> Should render correctly 1`] = `
<div>
  <div
    class="table-responsive"
  >
    <table
      class="table table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            File Name
          </th>
          <th>
            Text
          </th>
          <th>
            Number
          </th>
          <th>
            Hex
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            test2.csv
          </td>
          <td>
            uzUVbPfNKGawHWZHeqiO
          </td>
          <td>
            777459
          </td>
          <td>
            1ec4c6a048194c9b72fad4b0273f2533
          </td>
        </tr>
        <tr>
          <td>
            test2.csv
          </td>
          <td>
            angel
          </td>
          <td>
            888
          </td>
          <td>
            4f9e5d1c3a7b2eca8fb3456789012345
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
